File: Queue & Stack/Python/remove_all_adjacent_duplicates_in_string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        res = list()
        for a in s:
            if not res: res.append(a)
            else:
                if res[-1] == a: res.pop()
                else: res.append(a)
        return "".join(res)

File: Queue-Stack/CPP/remove_all_adjacent_duplicates_in_string.cpp
#include <string>

#include <vector>

using namespace std;

// Solution 1: Using two pointers to keep track of unique characters and remove duplicate

class Solution {
public:
    string removeDuplicates(string s) {
        vector<char> stk;
        for (char a: s) {
            if (stk.size() == 0) {
                stk.push_back(a);
            }
            else {
                if (stk.back() == a) stk.pop_back();
                else stk.push_back(a);
            }
        }
        return string(stk.begin(), stk.end());
    }
};